src/status_im/chat/commands/protocol.cljs

```diff
@@ -29,21 +29,36 @@
     and `cofx` map as argument, returns either `nil` meaning that no errors were
     found and command send workflow can proceed, or one/more errors to display.
     Each error is represented by the map containing `:title` and `:description` keys.")
-  (yield-control [this parameters cofx]
-    "Optional function, which if implemented, can step out of the normal command
-    workflow (`validate-and-send`) and yield control back to application before sending.
-    Useful for cases where we want to use command input handling (parameters) and/or
-    validating, but we don't want to send message before yielding control elsewhere.")
-  (on-send [this message-id parameters cofx]
+  (on-send [this command-message cofx]
     "Function which can provide any extra effects to be produced in addition to 
     normal message effects which happen whenever message is sent")
   (on-receive [this command-message cofx]
     "Function which can provide any extre effects to be produced in addition to
     normal message effects which happen when particular command message is received")
-  (short-preview [this command-message cofx]
+  (short-preview [this command-message]
     "Function rendering the short-preview of the command message, used when
     displaying the last message in list of chats on home tab.
     There is no argument names `parameters` anymore, as the message object
     contains everything needed for short-preview/preview to render.")
-  (preview [this command-message cofx]
+  (preview [this command-message]
     "Function rendering preview of the command message in message stream"))
+
+(defprotocol Yielding
+  "Protocol for defining commands yielding control back to application during the send flow"
```

Comment:
Maybe rename `send` -> `command` in the comment here.  I know it's only actually used for `/send` now, but just to keep it generic.

Reply:
It's meant as "command (any) send flow", with send flow being validation/yielding-control/sending-message. 
But it's little bit confusing, I agree :)

Reply:
Yeah, I get it. "send" as in "send command message". Yeah, confusing.